interpreter/golang/golang.go

```diff
@@ -0,0 +1,157 @@
+	pin       runtime.Pinner
+}
+
+func Loader(_ interpreter.EbpfHandler, info *interpreter.LoaderInfo) (
+	interpreter.Data, error) {
+	ef, err := info.GetELF()
+	if err != nil {
+		return nil, err
+	}
+	if !ef.IsGolang() {
+		return nil, nil
+	}
+
+	exec := info.FileName()
+	if len(exec) < 2 {
+		// There are cases where FileName() just returns /.
+		// In these cases we can not access the backing executable
+		// by FileName() and therefore can not continue here.
+		return nil, nil
+	}
+
+	return &golangData{
+		exec: exec,
+	}, nil
+}
+
+func (g *golangData) Attach(_ interpreter.EbpfHandler, pid libpf.PID,
+	_ libpf.Address, _ remotememory.RemoteMemory) (interpreter.Instance, error) {
+	gi := &golangInstance{}
+
+	executablePath := C.CString(fmt.Sprintf("/proc/%d/root%s", pid, g.exec))
+	defer C.free(unsafe.Pointer(executablePath))
+
+	//nolint:gocritic
+	status := C.symblib_goruntime_new(executablePath, &gi.goRuntime)
+	if status != C.SYMBLIB_OK {
+		return nil, fmt.Errorf("failed to create point resolver for '%s': %d",
+			C.GoString(executablePath), status)
+	}
+	gi.pin.Pin(unsafe.Pointer(&gi.goRuntime))
+
+	return gi, nil
+}
+
+func (g *golangInstance) GetAndResetMetrics() ([]metrics.Metric, error) {
+	return []metrics.Metric{
+		{
+			ID:    metrics.IDGolangSymbolizationSuccess,
+			Value: metrics.MetricValue(g.successCount.Swap(0)),
+		},
+		{
+			ID:    metrics.IDGolangSymbolizationFailure,
+			Value: metrics.MetricValue(g.failCount.Swap(0)),
+		},
+	}, nil
+}
+
+// Detach is a NOP for Golang.
+func (g *golangInstance) Detach(_ interpreter.EbpfHandler, _ libpf.PID) error {
+	if g.goRuntime != nil {
+		g.pin.Unpin()
+		C.symblib_goruntime_free(g.goRuntime)
+		g.goRuntime = nil
+	}
+	return nil
+}
+
+func (g *golangInstance) Symbolize(symbolReporter reporter.SymbolReporter, frame *host.Frame,
+	trace *libpf.Trace) error {
+	if !frame.Type.IsInterpType(libpf.Native) {
+		return interpreter.ErrMismatchInterpreterType
+	}
+	sfCounter := successfailurecounter.New(&g.successCount, &g.failCount)
+	defer sfCounter.DefaultToFailure()
+
+	if g.goRuntime == nil {
+		return errors.New("point resolver is out of scope")
+	}
+
+	var symbols *C.SymblibSlice_SymblibResolvedSymbol
+	defer C.symblib_slice_symblibresolved_symbol_free(symbols)
+
+	//nolint:gocritic
+	status := C.symblib_point_resolver_symbols_for_pc(g.goRuntime,
+		C.uint64_t(frame.Lineno), &symbols)
+	if status != C.SYMBLIB_OK {
+		return fmt.Errorf("failed to do point lookup at 0x%x: %d",
+			frame.Lineno, status)
+	}
+
+	// Access resolved symbols
+	symbolsSlice := unsafe.Slice((*C.SymblibResolvedSymbol)(unsafe.Pointer(symbols.data)),
+		symbols.len)
+	if len(symbolsSlice) != 1 {
+		return fmt.Errorf("unexpected return for point lookup: %d", len(symbolsSlice))
+	}
+
+	frameID := libpf.NewFrameID(libpf.NewFileID(uint64(frame.File), uint64(frame.File)),
+		libpf.AddressOrLineno(symbolsSlice[0].line_number))
+
+	trace.AppendFrameID(libpf.GolangFrame, frameID)
```

Comment:
Should this provide the mapping information similar as the non-symbolized native code is done?

Reply:
At this point Go frames are treated as interpreter and no longer as native code. So I guess the answer here is no, and mapping information is not provided similar to other native code.